### PR TITLE
feat: add `airflow.kubernetesPodTemplate.extraInitContainers`

### PR DESCRIPTION
**Pull request by jmcgrath207**
## What issues does your PR fix?

N/A

## What does your PR do?

- Adds the `airflow.kubernetesPodTemplate.extraInitContainers` value to specify extra KubernetesExecutor init-containers

### For all Pull Requests

- [X] Commits are [signed off](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#sign-your-work)
- [X] Commits have [semantic messages](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#semantic-commit-messages)
- [X] Documentation [updated](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#documentation)
- [X] Passes [ct linting](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#linting)

### For releasing ONLY

- [ ] Chart.yaml [version bumped](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#versioning)
- [ ] CHANGELOG.md updated

### CODE REVIEW

**Comment by jmcgrath207** [comment]
Hey @thesuperzapper @gsemet  , is there anything else I need to do to get this merged?

File: charts/airflow/files/pod_template.kubernetes-helm-yaml
Comment (thesuperzapper) [inline charts/airflow/files/pod_template.kubernetes-helm-yaml:51]: You also need to update line 41, so that it will enable the `initContainers` section if `.Values.kubernetesPodTemplate.extraInitContainers` is defined. (Otherwise this wont work unless `$extraPipPackages` or `.Values.dags.gitSync.enabled` are enabled)

For example:
```yaml
  {{- if or ($extraPipPackages) (.Values.dags.gitSync.enabled) (.Values.kubernetesPodTemplate.extraInitContainers) }}
  initContainers:
```

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
@jmcgrath207 thanks for the PR, I have left a small fix that is needed before we can merge.

File: charts/airflow/values.yaml
Comment (thesuperzapper) [inline charts/airflow/values.yaml:319]: nit: can you make this consistent with the other docscrings in the section:

```yaml
## extra init-containers to run in the Pod template
```

Comment (jmcgrath207) [reply charts/airflow/values.yaml]: ah makes sense. I fixed it

File: charts/airflow/files/pod_template.kubernetes-helm-yaml
Comment (jmcgrath207) [reply charts/airflow/files/pod_template.kubernetes-helm-yaml:51]: Ah gotcha, this is resolved.

**Review by jmcgrath207 — COMMENTED** [review]
changes complete. Waiting for approval

**Comment by thesuperzapper** [comment]
@jmcgrath207 thanks for your work! 

I will put this in the 8.6.0 milestone (as it adds a new value).

File: charts/airflow/files/pod_template.kubernetes-helm-yaml
Comment (thesuperzapper) [inline charts/airflow/files/pod_template.kubernetes-helm-yaml:41]: This is incorrect, the value is `airflow.kubernetesPodTemplate.extraInitContainers`, not `kubernetesPodTemplate.extraInitContainers`.

File: charts/airflow/files/pod_template.kubernetes-helm-yaml
Comment (thesuperzapper) [inline charts/airflow/files/pod_template.kubernetes-helm-yaml:49]: This is incorrect, the value is `airflow.kubernetesPodTemplate.extraInitContainers`, not `kubernetesPodTemplate.extraInitContainers`.

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
@jmcgrath207 I just realized you were not correctly referencing the value you created.

The value is `airflow.kubernetesPodTemplate.extraInitContainers`, not `kubernetesPodTemplate.extraInitContainers`.

**Comment by thesuperzapper** [comment]
@jmcgrath207 do you still have time to apply the changes needed to address comments in https://github.com/airflow-helm/charts/pull/446#pullrequestreview-832165840?

**Review by thesuperzapper — APPROVED** [review]
@jmcgrath207 I have gone ahead and fixed your PR, so it can be merged for 8.6.0.

**Pull request merged**
